### ISSUE TITLE
[gatsby-source-wordpress] Can't fetch the featured_media field

### ISSUE THREAD

**Issue by ShawnBaek** [gatsbyjs/gatsby#12237]
## Description
After upgrade to the latest GatsbyJS, I can't fetch some wordpress.com REST API fields.

`author` `featured_media` is not working.

## "gatsby-source-wordpress": "^3.0.2"

I rechecked 3.0.2. It has no problem. 
The problem has only happened in the latest version.

```javascript

export const blogQuery = graphql`
  query {
    allWordpressPost(sort: { fields: [date] }) {
      edges {
        node {
          id
          title
          excerpt
          slug
          date(formatString: "MMMM DD, YYYY")
          categories {
            id
            name
          }
          tags {
            id
            name
          }
          author {
            id
            url
            name
            description
            avatar_urls {
              wordpress_24
              wordpress_48
              wordpress_96
            }
          }
          featured_media {
            localFile {
              childImageSharp {
                # edit the maxWidth value to generate resized images
                resolutions(width: 500, height: 500) {
                  src
                  width
                  height
                  srcSet
                  aspectRatio
                }
              }
            }
            id
            media_details {
              width
              height
              file
              filesize
            }
            source_url
          }
          jetpack_related_posts {
            wordpress_id
            url
            title
            date
            format
            excerpt
            rel
            context
          }
          ...PostIcons
        }
      }
    }
  }
`
```

### Steps to reproduce

I checked **Wordpress.com** `REST API` using Postman.

The `featured_media` field and `author` object are fetched.
 
<img width="859" alt="screen shot 2019-03-02 at 12 31 11 pm" src="https://user-images.githubusercontent.com/12643700/53676526-26592880-3ce7-11e9-8e84-d98a9dae008f.png">
<img width="769" alt="screen shot 2019-03-02 at 12 29 44 pm" src="https://user-images.githubusercontent.com/12643700/53676527-26592880-3ce7-11e9-80a3-e410309bd70e.png">

### Expected result

All wordpress datas should be fetched.

### Actual result

<img width="673" alt="screen shot 2019-03-02 at 12 27 52 pm" src="https://user-images.githubusercontent.com/12643700/53676506-9b782e00-3ce6-11e9-8298-851219b6a085.png">

<img width="1528" alt="screen shot 2019-03-03 at 7 08 56 pm" src="https://user-images.githubusercontent.com/12643700/53693750-f5145180-3de7-11e9-9417-3d432cd0958b.png">

**Comment by DSchau** [comment]
Did your credentials change? Did the way you sourced variables (e.g. username/pw) change?

You can see a number of 401 errors in your console, so the issue is that there aren't any nodes being created because the plugin is failing to fetch _any_ content with the aforementioned 401 error.

If you're able, I'd strongly encourage you to [create a reproduction](https://gatsby.dev/reproduction), so that we can most effectively debug this.

Thanks for your time, and sorry for the issue you're running into!

**Comment by ShawnBaek** [comment]
No, I didn't change my account information.

I tested `gatsby-source-wordpress@3.0.2` It works perfectly. 

This problem only happened in the latest version.

**Comment by pieh** [comment]
Looking at logs, seems like there is regression causing wordpress plugin to make unauthenticated requests to `media` endpoint (wp.com hosting requires those to be authenticated). We need response from that to download images and link `featured_media` field to actual files.

**Comment by ryuheiyokokawa** [comment]
Having the same regression here.  Using normal WP 5.1 (self-hosted).  Can confirm that the media asset exists via REST API, but doesn't show on graphql side (FYI, all media has been unattached).  Seems to only affect featured_media for me, but unsure as that is where it got stuck first.  I've also tagged at 3.0.2 to prevent issues for now.  Thanks @ShawnBaek !

**Comment by ShawnBaek** [comment]
Is it solved?

**Comment by inadeqtfuturs** [comment]
Having a similar issue on `gatsby-source-wordpress@3.0.54` where I can hit the routes on the wordpress console, but getting the following in terminal on develop:

```
Path: /wp/v2/sites/katiedyson91140482.wordpress.com/media?per_page=100&page=1
The server response was "401 Unauthorized"
Inner exception message: "That API call requires authentication against the correct blog."
```

In addition to getting the same kind of path errors, I'm also getting the following before the wordpress-plugin fires:

```
Path: /oauth2/token
The server response was "400 Bad Request"
```

**Comment by avinoamsn** [comment]
I'm having the same issue, @ShawnBaek did you ever figure out what the problem was?

**Comment by vkallore** [comment]
No the problem still exists. Using `"gatsby-source-wordpress": "^3.1.13"`!!

**Comment by wardpeet** [comment]
Is it possible to create a reproduction for us? Or tell us how to repro this? If not, there is no real point of re-opening the issue.

**Comment by vkallore** [comment]
Well, I've replicated the code. Please let us know how to get the `featured_image` from [here](https://fdou3.sse.codesandbox.io/___graphql?query=query%20MyQuery%20%7B%0A%20%20allWordpressPost%20%7B%0A%20%20%20%20edges%20%7B%0A%20%20%20%20%20%20node%20%7B%0A%20%20%20%20%20%20%20%20id%0A%20%20%20%20%20%20%20%20slug%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%0A). Please let me know if the [implementation](https://codesandbox.io/s/gatsby-starter-default-fdou3) is wrong.

**Comment by avinoamsn** [comment]
Thanks @vkallore. Running the same setup with version 3.0.2 doesn't seem to remedy the problem, as it did for @ShawnBaek.

**Comment by vkallore** [comment]
@avinoamsn The blog which I am working on my local has featured image. _WP_SITE/wp-json/wp/v2/posts_ responses with `"featured_media": INT_MEDIA_ID`.

Even the [WP API demo website API](https://demo.wp-api.org/wp-json/wp/v2/posts) responds with `featured_media`, though they don't have one for their post (its 0).

**Comment by ndotmdot** [comment]
I had the same issue. Running the WP-Optimize plugin with default setting and `gatsby clean `afterwards solved it for me

**Comment by alex789** [comment]
I tried downgrading to gatsby-source-wordpress 3.0.2 as mentioned above but didn't work.

I then unpublished a page which had no featured image and this solved the issue.

**Comment by riddla** [comment]
For others hitting the missing `featured_media` field: I got some input from an old Wordpress API issue, https://github.com/WP-API/WP-API/issues/2596.
I tried to get the featured media directly via the REST API of my WordPress installation (`/wp-json/wp/v2/media/13`) and got a 403. Using another image then solved it for me.

There seem to be some weird bugs regarding the media endpoint. From what I can tell i stumbled upon the following:

> This issue also occurs if you attach the featured image at draft stage (the post has never been published). The post parent of the attachment retains the post parent id of a revision, which means the same permissions problem occurs.
(https://core.trac.wordpress.org/ticket/41445#comment:24)

Maybe when https://core.trac.wordpress.org/ticket/41445 lands in WordPress `5.4` those kind of errors will be gone.
